### PR TITLE
Create LocalExperimentTracker for storing information in local locations

### PR DESCRIPTION
**Pull request by melmatlis**
The code generates a new directory per experiment and enables storing the following information:
1. Store a confusion matrix as a comma separated file
2. Store a serialized version of model errors for down stream analysis. The file is stored as a Pickle object

### CODE REVIEW

File: presidio_evaluator/experiment_tracking/local_tracker.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+                         "experiment_params.json")
+        print(f"saving experiment data to directory {self.experiment_name}")
+        with file_path.open('w') as json_file:
+            json.dump(self.__dict__, json_file)
+
+
+if __name__ == "__main__":
+
+    directory = Path(Path.cwd(), 'experiments')
+    experiment_name = "my_first_experiment"
```
Comment (tranguyen221) [inline presidio_evaluator/experiment_tracking/local_tracker.py:84]: The code looks great to me. Just a minor comment. Should we add the datetime in the experiment name, so the user can track when the experiment is done?

**Review by tranguyen221 — COMMENTED** [review]
The code looks great to me. Just add a minor comment.

File: presidio_evaluator/experiment_tracking/local_tracker.py (hunk 1 above)
Comment (melmatlis) [reply presidio_evaluator/experiment_tracking/local_tracker.py:84]: It's a great idea, I will add it as a parameter in the `.json` file, this way the experiment_name will remain clear and easy to remember but the the date time of the experiment execution will be stored. What do you think?

Comment (tranguyen221) [reply presidio_evaluator/experiment_tracking/local_tracker.py:84]: I think that would be a great option. Just a question, if a user runs the same experiment name but in 2 different timestamps, the old results will be overwritten by the new one?

Comment (melmatlis) [reply presidio_evaluator/experiment_tracking/local_tracker.py:84]: At the moment yes.

Comment (tranguyen221) [reply presidio_evaluator/experiment_tracking/local_tracker.py:84]: Ok. In this case, I think we add datetime in the .json file.

Comment (melmatlis) [reply presidio_evaluator/experiment_tracking/local_tracker.py:89]: done

Comment (tranguyen221) [reply presidio_evaluator/experiment_tracking/local_tracker.py:89]: Great! Thank Mel!

**Review by tranguyen221 — APPROVED** [review]

**Pull request merged**